### DISCUSSION TITLE
Where is GPC 20070904?

### DISCUSSION THREAD

**Discussion by blakemcbride** [samiam95124/Pascal-P6#51] [General]
The build instructions say to start with GPC 20070904.  I think GPC is pretty dead.  In fact, the only version I could even find is 20060325!  I'd like to monkey around with P6, but I'll need a better way of bootstrapping it.  Is there another way?

**Comment by samiam95124** [comment]
On 2023-04-13 20:03, Blake McBride wrote:
> The build instructs say to start with GPC 20070904. I think GPC is
> pretty dead. In fact, the only version I could even find is 20060325!
> I'd like to monkey around with P6 but I'll need a better way of
> bootstrapping it. Is there another way?
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe
> [2].
> You are receiving this because you are subscribed to this
> thread.Message ID:
> ***@***.***>
> 
> Links:
> ------
> [1] https://github.com/samiam95124/Pascal-P6/discussions/51
> [2] 
> https://github.com/notifications/unsubscribe-auth/AEH6RWRMOH3A4T2HLLUWB23XBC5BXANCNFSM6AAAAAAW53TKUY

Its on the mirror site:

https://sourceforge.net/projects/pascal-p6/files/

Yes, its getting hard to find, that's why I am pushing for P6 to 
self-bootstrap.
Its the Linux version of GPC. There is a windows 32 bit version. See the 
P6 docs for how to set it up.

Scott Franco

**Comment by samiam95124** [comment]
On 2023-04-13 20:30, ***@***.*** wrote:
> On 2023-04-13 20:03, Blake McBride wrote:
>> The build instructs say to start with GPC 20070904. I think GPC is
>> pretty dead. In fact, the only version I could even find is 20060325!
>> I'd like to monkey around with P6 but I'll need a better way of
>> bootstrapping it. Is there another way?
>> 
>> --
>> Reply to this email directly, view it on GitHub [1], or unsubscribe
>> [2].
>> You are receiving this because you are subscribed to this
>> thread.Message ID:
>> ***@***.***>
>> 
>> Links:
>> ------
>> [1] https://github.com/samiam95124/Pascal-P6/discussions/51
>> [2] 
>> https://github.com/notifications/unsubscribe-auth/AEH6RWRMOH3A4T2HLLUWB23XBC5BXANCNFSM6AAAAAAW53TKUY
> 
> Its on the mirror site:
> 
> https://sourceforge.net/projects/pascal-p6/files/
> 
> Yes, its getting hard to find, that's why I am pushing for P6 to 
> self-bootstrap.
> Its the Linux version of GPC. There is a windows 32 bit version. See
> the P6 docs for how to set it up.
> 
> Scott Franco

Humm, I'll bet that was confusing. The Linux version at the above site 
is 64  bit.
The Windows 32 bit is out there, but its been a while since I used that, 
so don't know
where it is at the moment.

Scott

**Comment by samiam95124** [comment]
Thanks to Blake Mcbride for posting this.

I use the 64 bit linux version under Ubuntu 20.04lts. It needed a special option to work, but I believe that is in the GPC makefile.
The last windows version I used was 32 bit, and needed to be matched with a mingw and gcc of the same version. Its been at least a year since I last ran it.

The plan is to replace GPC with self-booting Pascal-P6, using the pgen module. This work is quite far along, it has already printed a "hello world" program. I put that aside temporarily to work on the 0.2 version, the reason being that there were issues with P6 that I preferred to clean up rather than design workarounds in P6 for them that I would just have to remove later, doubling the workload.

Thus the plan is:

0.2 - The "clean up" version.
0.3 - Working pgen, including self-bootstrapping.
0.4 - Tie in to the Petit-ami library.

Thanks,

Scott Franco

**Comment by samiam95124** [comment]
A quick search found the 32 bit Windows version for mingw here:

https://www.gnu-pascal.de/binary/mingw32/
